felles/src/main/kotlin/no/nav/familie/kontrakter/felles/Ressurs.kt

```diff
@@ -47,6 +47,13 @@ data class Ressurs<T>(
         }
     }
 
+    fun getDataOrThrow(): T {
+        return when (this.status) {
+            Status.SUKSESS -> data ?: error("Data er null i Ressurs")
+            else -> error(melding)
```

Comment:
Hva tenker dere om denne meldingen? Nå putter vi random ting i den, men den var egentlig tenkt til funksjonelle meldinger til frontend.

Reply:
Det er vel den bruken jeg også har sett for meg, men når den først er der kan vi jo sørge for at den logges. Det kan jo tenkes at den sparer oss litt graving ved feilsøking også på interapplikasjonskall.

Reply:
mhm, men hva om vi legger til funksjonellFeilmelding som kan brukes til presentasjon frontend (og kan inneholde sensitive data). Denne kan automatisk logges til secureLogger.

Så kan vi bruke melding til en generell melding uten sensitive data og som alltid kan logges. Det kan være suksess og feilmelding.
Vi setter også stacktrace og denne kan også logges til secureLogger?

Reply:
Og kanskje lagt til en toSecureString() som man opplyser teamet om å bruke?

Reply:
Burde ikke stacktrace være logget på avsendersiden? Redd for at loggen kan bli slitsom å lese hvis stack trace som logges i en applikasjon inneholder stack trace fra en annen appliaksjon.

Reply:
Jo, den kan gjerne logges på avsendersiden, men fint at den sendes med til mottaker slik at vi kan plukke den opp i familie-prosessering 👍

Reply:
Hvis formålet egentlig er en funksjonell "feilmelding" burde vi enten bruke den til det eller rename (legge til nytt attributt + fjerne dette når man har flyttet over).

Reply:
Helt enig, men så gir det også mening å ha en mer teknisk feilmelding. Derfor foreslår jeg å beholde melding (til generelle tekniske feil/suksess meldinger) og legge til funksjonellMelding/funksjonellFelmelding. Vi har ikke brukt melding så aktiv mot frontend uansett, så er ikke så stress å gå over og endre det.

Reply:
Når vi, som du skriver Henning, putter random ting i den mens den opprinnelig var ment for noe mer spesifikt (og kanskje blir det igjen) så tenker jeg at det kanskje er lurt å gi den et mer dekkende navn. Vi hadde en lignende problemstilling for exception-håndtering i integrasjoner der vi endte med å legge til et felt som tydelig kommuniserer at her skal sensitiv informasjon plasseres. Så er det exception-handleren (ApiExceptionHandler) som faktisk bruker feltet og skriver det til secure logger.
`public OppslagException(String message, String kilde, Level level, HttpStatus httpStatus, Throwable error, String sensitiveInfo)`

Reply:
Gode poenger. 

Når vi legger inn `funksjonellFeilmelding` tror jeg det kan være lurt å ta en `ætt channel` på slack og informere alle om hvordan vi ønsker å bruke disse

Reply:
Ja, jeg kan invitere til en kaffe og ressurs-prat på mandag 👍 
Vil du legge inn feltet i kontrakten, samt toSecureString()?

Reply:
Har man tenkt å innføre tjenester som returnerer 2xx ved feil da eller?

Reply:
Det er vel litt på siden av denne diskusjonen tenker jeg. Vi trenger ikke det, men for ressurser som går til frontend mener jeg at vi burde returnere 200 ok med funksjonellFeilmelding satt eksplisitt ja 👍